src/nu/validator/htmlparser/io/Encoding.java

```diff
@@ -254,9 +254,7 @@ public static String toNameKey(String str) {
             if (c >= 'A' && c <= 'Z') {
                 c += 0x20;
             }
-            if (!((c >= '\t' && c <= '\r') || (c >= '\u0020' && c <= '\u002F')
-                    || (c >= '\u003A' && c <= '\u0040')
-                    || (c >= '\u005B' && c <= '\u0060') || (c >= '\u007B' && c <= '\u007E'))) {
+            if (!Arrays.asList('\t','\n','\f','\r','\u0020').contains(c)) {
```

Comment:
The performance characteristics of `Arrays.asList` are a bit unclear to me, but it seems bad to call it again on each loop iteration. I think `(c == ' ' || c == '\t' || c == '\n' || c == '\f' || c == '\r')` would probably be the best thing to do here.

Reply:
> The performance characteristics of `Arrays.asList` are a bit unclear to me, but it seems bad to call it again on each loop iteration. I think `(c == ' ' || c == '\t' || c == '\n' || c == '\f' || c == '\r')` would probably be the best thing to do here.

Thanks, yeah, I should have just done it that way to begin with. Anyway, changed it to that now. Amended to the existing commit, and force-pushed to the branch — so that we just have one commit here, and so need to squash before merging.